app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/blocklist/AppTrackerListUpdateWorker.kt

```diff
@@ -73,6 +82,11 @@ class AppTrackerListUpdateWorker(context: Context, workerParameters: WorkerParam
                     vpnDatabase.vpnAppTrackerBlockingDao().getTrackerBlocklistMetadata()?.eTag
                 val updatedEtag = blocklist.etag.value
 
+                if (updatedEtag == currentEtag) {
```

Comment:
I know I initially suggested this and I think we should keep this to ensure no other race conditions. But I wonder if we should still try and do a one-time fix for users who have gotten into a mismatched state via a small DB migration where we clear the metadata?

Reply:
will do